agent_patch_distr.py

```diff
@@ -0,0 +1,245 @@
+import sys
+import os
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.colors
+import json
+import math
+from abm.replay.replay import ExperimentReplay
+
+########################### DOCUMENTATION ######################################
+#
+# What agent_patch_distr.py does:
+# When doing an experiment with different values for RADIUS_RESOURCE the script creates
+# for each value a 2D heatmap plot of the exploitable area (area covered by patches)
+# and a plot of the locations of the centers of the patches for testing reasons. The
+# plots contain the parameter values in the respective filename and the plot title itself.
+# Furthermore normalized_pixel_array.npz is created for doing quick adaptations in the plots.
+# The scipt can also handles the case if N=0 and further allows plot the experiment data
+# generated by nb_exp13_small_patch_only.py
+
+#
+# How to use the scripts_
+# - run nb_exp12.py or nb_exp13_small_patch_only.py (define EXPERIMENT_NAME, e.g. patch_place_distr)
+# - adapt nb_exp12.py by varying `num_patches`. Define at least two values, e.g. `num_patches = [1, 5]`
+#   agent_patch_distr.py fails if only one value is given!
+#   adapt nb_exp13_small_patch_only.py by varying `N` and/or `num_batches`
+# - adapt filepath in agent_patch_distr.py (e.g. filepath = 'home/ABM/abm/data/simulation_data/patch_place_distr')
+# - run agent_patch_distr.py (the directory where to patch_exploitable_area.py is arbitrary)
+# - a folder patch_place_distr/summary/agent_patch_distr will be created containing the plots named after
+#   the the different parameters used and the .npz-files for redoing the plots
+#
+################################################################################
+
+# TODO see modifiy_pixel_array_circle()
+
+# for loading data from /summary
+filepath = '/ABM/abm/data/simulation_data/agent_patch_place_distr/'
+# for storing plots of patch distribution
+folderpath = filepath + '/summary/agent_patch_distr'
+window_pad = 30 # hardcoded
+
+# see "Equation of a circle" in  https://en.wikipedia.org/wiki/Circle#Equations
+def modifiy_pixel_array_circle(pixel_array, m_x, m_y, radius):
+    """ Modifies a 2D pixel array such that in the area of a circle with given
+    position (center of the circle) and radius a one is added.
+
+    Parameters
+    ----------
+    pixel_array : 2d array
+    m_x : x-location of circle.
+    m_y : y-location of circle.
+    radius : radius of circle.
+
+    Returns: modified pixel_array
+    """
+    ENV_HEIGHT, ENV_WIDTH = np.shape(pixel_array)
+    # TODO Check if arena consists of pixel id = 0 until pixel id = ENV_WIDTH-1 or
```

Comment:
Is this TODO item finished? If so, remove it, if not, we have to decide if it has to be done or can be skipped.

Reply:
This is of minor relevance and can be skipped.